dbms/src/Storages/DeltaMerge/FilterParser/FilterParser.cpp

```diff
@@ -398,15 +398,15 @@ RSOperatorPtr FilterParser::parseDAGQuery(const DAGQueryInfo & dag_info,
 
     if (dag_info.filters.size() == 1)
     {
-        op = cop::tryParse(*dag_info.filters[0], columns_to_read, creator, dag_info.timezone_info, log);
+        op = cop::tryParse(dag_info.filters[0], columns_to_read, creator, dag_info.timezone_info, log);
     }
     else
     {
         /// By default, multiple conditions with operator "and"
         RSOperators children;
-        for (size_t i = 0; i < dag_info.filters.size(); ++i)
+        for (int i = 0; i < dag_info.filters.size(); ++i)
```

Comment:
why change this?
```suggestion
        for (size_t i = 0; i < dag_info.filters.size(); ++i)
```

Reply:
You can see the generated code from protobuf. The return value of `RepeatedPtrField::size` is `int`.

Reply:
indeed